container-search/src/main/java/com/yahoo/search/grouping/GroupingRequest.java

```diff
@@ -151,14 +151,8 @@ public static GroupingRequest newInstance(Query query) {
      * @return The list of grouping requests.
      */
     @SuppressWarnings({ "unchecked" })
+    @Deprecated
     public static List<GroupingRequest> getRequests(Query query) {
-        Object lst = query.properties().get(PROP_REQUEST);
-        if (lst == null) {
-            return Collections.emptyList();
-        }
-        if (!(lst instanceof List)) {
-            throw new IllegalArgumentException("Expected " + GroupingRequest.class + ", got " + lst.getClass() + ".");
-        }
-        return (List<GroupingRequest>)lst;
+        return query.getSelect().getGrouping();
```

Comment:
You also need to change newRequest to access query.getSelect().getGrouping() in the same way.

Reply:
Did you mean newInstance?